packages/core/src/memory/types.ts

```diff
@@ -56,6 +57,12 @@ export type MemoryConfig = {
       };
   threads?: {
     generateTitle?: boolean;
+    generateTitleInstructions?:
+      | string
+      | ((
+          userMessage: CoreUserMessage | undefined,
+          { runtimeContext }: { runtimeContext: RuntimeContext },
+        ) => Promise<string>);
```

Comment:
I think we should do something like `generateTitle?: boolean | { enabled: boolean, instructions: string | Fn }`